filesetup.c

```diff
@@ -837,8 +837,13 @@ uint64_t get_start_offset(struct thread_data *td, struct fio_file *f)
 	if (o->file_append && f->filetype == FIO_TYPE_FILE)
 		return f->real_file_size;
 
-	return td->o.start_offset +
-		td->subjob_number * td->o.offset_increment;
+	if (o->start_offset_percent > 0) {
+		return (f->real_file_size * o->start_offset_percent / 100) +
+			(td->subjob_number * td->o.offset_increment);
```

Comment:
@cbwest3 Can you round this to blockalign if it is set or to the minimum block size if it is not?